nexus/src/app/instance.rs

```diff
@@ -1610,11 +1610,19 @@ impl super::Nexus {
         if let Some(max_bytes) = params.max_bytes {
             request = request.max_bytes(max_bytes);
         }
-        if let Some(from_start) = params.from_start {
-            request = request.from_start(from_start);
-        }
-        if let Some(most_recent) = params.most_recent {
-            request = request.most_recent(most_recent);
+        match (params.from_start, params.most_recent) {
+            (Some(from_start), None) => {
+                request = request.from_start(from_start);
+            }
+            (None, Some(most_recent)) => {
+                request = request.most_recent(most_recent);
+            }
+            _ => {
+                return Err(Error::invalid_request(
+                    "Exactly one of 'from_start' \
+                    or 'most_recent' must be specified.",
+                ));
```

Comment:
don't super love this, and maybe @hawkw's new dropshot error type stuff can help? this reproduces a check that's done in both Propolis and propolis-sim. my thinking is, the connection to Propolis _could_ error for other reasons, and whatever that is, it is correctly categorized as a 500. i didn't want to just forward any 400 error from Propolis out to users here because Propolis errors aren't necessarily intended for direct end-user consumption...

if parameter validation gets more strict on the Propolis side, though, things checked there and not here will be 500s..